.github/workflows/build.yml

```diff
@@ -117,6 +128,8 @@ jobs:
         with:
           macos-skip-brew-update: 'true'
       - name: Build docs
+        env:
+          SPHINX_GITHUB_CHANGELOG_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I can only [set `permissions` of the `GITHUB_TOKEN`](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) per job or per workflow. Setting the token's permissions to read-only for this entire job will likely cause failure in the upload to gh-pages step.

To ensure read-only access to the changelog ext, I think A separate key may be more needed. The ext does nothing if there is no token to use.

Reply:
I repealed the changelog generation as it can be addressed separately (if desired).